server/routes/daily-attendance-summary/handlers/attendance.ts

```diff
@@ -0,0 +1,210 @@
+    const uniqueActivities = attendances.map(c => c.activitySummary).filter((v, k, arr) => arr.indexOf(v) === k)
+
+    if (
+      !attendanceSummaryFilters ||
+      attendanceSummaryFilters.categoryFilters.length === 0 ||
+      !attendanceSummaryFilters.activityFilters ||
+      attendanceSummaryFilters.activityFilters.length === 0
+    ) {
+      attendanceSummaryFilters = defaultFilters(activityDate, '', uniqueCategories, uniqueActivities)
+      req.session.attendanceSummaryFilters = attendanceSummaryFilters
+    }
+    const categoryFilters = attendanceSummaryFilters.categoryFilters
+      .filter((category: FilterItem) => category.checked === true)
+      .map((category: FilterItem) => category.value)
+
+    const activityFilters = attendanceSummaryFilters.activityFilters
+      .filter((activity: FilterItem) => activity.checked === true)
+      .map((activity: FilterItem) => activity.value)
+
+    const prisonerNumbers = attendances
+      .filter(a => categoryFilters.includes(a.categoryName) && activityFilters.includes(a.activitySummary))
+      .map(a => a.prisonerNumber)
+
+    const unfilteredAttendees =
+      prisonerNumbers?.length > 0
+        ? await this.prisonService.searchInmatesByPrisonerNumbers(prisonerNumbers, user).then(inmates =>
+            inmates.map(i => ({
+              name: `${i.firstName} ${i.lastName}`,
+              prisonerNumber: i.prisonerNumber,
+              location: i.cellLocation,
+              attendance: attendances
+                .filter(a => categoryFilters.includes(a.categoryName) && activityFilters.includes(a.activitySummary))
+                .find(a => a.prisonerNumber === i.prisonerNumber),
+            })),
+          )
+        : []
+
+    const attendees = unfilteredAttendees.filter(
+      a =>
+        this.includesSearchTerm(a.name, attendanceSummaryFilters.searchTerm) ||
+        this.includesSearchTerm(a.prisonerNumber, attendanceSummaryFilters.searchTerm),
+    )
+
+    if (status === 'NotAttended') {
+      return res.render('pages/daily-attendance-summary/not-attended', {
+        activityDate,
+        status,
+        attendees,
+        attendanceSummaryFilters,
+      })
+    }
+
+    return res.render('pages/daily-attendance-summary/attended', {
+      activityDate,
+      status,
+      attendees,
+      attendanceSummaryFilters,
+    })
+  }
+
+  POST = async (req: Request, res: Response): Promise<void> => {
+    const { attendanceSummaryFilters } = req.session
+    const activityDate = req.query.date ? toDate(req.query.date as string) : undefined
+    const isoDateString = formatDate(new Date(activityDate), 'yyyy-MM-dd')
+    const { status } = req.query
+    attendanceSummaryFilters.searchTerm = req.body?.searchTerm ? (req.body?.searchTerm as string) : ''
+
+    req.session.attendanceSummaryFilters = parseFiltersFromPost(
+      attendanceSummaryFilters,
+      req.body?.categoryFilters ? req.body?.categoryFilters : undefined,
```

Comment:
Wouldn't `req.body.categoryFilters` either hold a value or be undefined? What's the value of the ternary operator here?

Reply:
Removed redundant code